### ISSUE TITLE
 "double free or corruption (out)" triggered by GpuMat::default()

### ISSUE THREAD

**Issue by 3togo** [twistedfall/opencv-rust#478]
I got errors when  running examples/cuda.rs. After some tracing, I found that whenever I initialize a variable using GpuMat::default(), it will trigger "double free or corruption (out)".  How to fix it?

Thanks in advance for any help.

To debug it, I simplified the code to:

```
use opencv::core::GpuMat;
use opencv::Result;
fn main() -> Result<()>{
    let img_gpu = GpuMat::default()?;
    println!("Done");
    println!("{:#?}", img_gpu);
    Ok(())
}
```

Below is the backtrace:
```
(gdb) bt
#0  __pthread_kill_implementation (no_tid=0, signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=6, threadid=<optimized out>) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=<optimized out>, signo=signo@entry=6) at ./nptl/pthread_kill.c:89
#3  0x00007ffff0e3c406 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#4  0x00007ffff0e2287c in __GI_abort () at ./stdlib/abort.c:79
#5  0x00007ffff0e235a4 in __libc_message (fmt=fmt@entry=0x7ffff0fb890f "%s\n") at ../sysdeps/posix/libc_fatal.c:150
#6  0x00007ffff0e9b147 in malloc_printerr (str=str@entry=0x7ffff0fbb7d8 "double free or corruption (out)") at ./malloc/malloc.c:5651
#7  0x00007ffff0e9d2c0 in _int_free (av=0x7ffff0ff6c80 <main_arena>, p=0x7ffff2f36600, have_lock=<optimized out>) at ./malloc/malloc.c:4574
#8  0x00007ffff0e9f773 in __GI___libc_free (mem=<optimized out>) at ./malloc/malloc.c:3367
#9  0x000055555557144d in cv_cuda_GpuMat_Allocator_delete(cv::cuda::GpuMat::Allocator*)
    (instance=0x7ffff2f36610 <(anonymous namespace)::cudaDefaultAllocator>)
    at /home/joe/gitee/j_misc/j_cv_demo/rust/eg001/target/debug/build/opencv-2a210dafba38eb22/out/core.cpp:9218
#10 0x000055555556f1b3 in opencv::opencv::hub::core::{impl#341}::drop (self=0x7fffffffd928)
    at target/debug/build/opencv-2a210dafba38eb22/out/opencv/core.rs:24545
#11 0x000055555556f17a in core::ptr::drop_in_place<opencv::opencv::hub::core::GpuMat_Allocator> ()
    at /rustc/8ede3aae28fe6e4d52b38157d7bfe0d3bceef225/library/core/src/ptr/mod.rs:497
#12 0x000055555556e14e in opencv::opencv::hub::core::GpuMat::default ()
    at /home/joe/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/opencv-0.84.1/src/manual/core/gpumat.rs:11
#13 0x000055555556edbb in eg001::main () at src/main.rs:4
```

**Comment by twistedfall** [comment]
I'll check it out, thanks for the report!

**Comment by twistedfall** [comment]
This problem should be fixed in v0.84.2

**Comment by 3togo** [comment]
Many thanks.
